pkg/kubelet/volumemanager/cache/desired_state_of_world.go

```diff
@@ -366,6 +466,15 @@ func (dsw *desiredStateOfWorld) PodExistsInVolume(
 		return false
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.ExpandInUsePersistentVolumes) {
+		if volumeObj.seLinuxFileLabel != seLinuxMountOption {
+			// The volume is in DSW, but with a different SELinux mount option.
+			// Report it as unused, so the volume is unmounted and mounted back
+			// with the right SELinux option.
+			return false
+		}
+	}
```

Comment:
When do we expect context is ASOW to be not same as one is desired state? If kubelet was down and another pod got created on same node with different selinux context?

Reply:
There are two cases:

* As you pointed out, kubelet restart + reconstruction + user created a new pod with a new context.
* User creates two pods with different contexts. The first pod will get to DSW + get mounted + to ASW. The second pod will be rejected by DSW as having a wrong context. Now, when the first pod is deleted, it leaves DSW, the second pod enters DSW, but something must unmount the volume and mount it again with a different context.

Reply:
I wonder if it should be written in such a way that - selinux_context + volume_name,  make for key in ASOW and DSOW maps, because that is our intention at the end of day.

Reply:
It will be problematic to see if a volume is already mounted with a different SELinux context. Reconciler will try to always mount such a volume and errors from `/bin/mount` about wrong SELinux contexts are ugly (`mount point not mounted or bad option`, without any note about SELinux).

Reply:
Yeah by default, it would be problematic. But for ReadWriteOnce pods, shouldn't another pod not try to mount same volume at the same time? (and it does well - mount error is a valid error).